hoss/dimension_utilities.py

```diff
@@ -418,27 +420,42 @@ def add_index_range(
     the antimeridian or Prime Meridian) will have a minimum index greater
     than the maximum index. In this case the full dimension range should be
     requested, as the related values will be masked before returning the
-    output to the user.
+    output to the user. When the dimensions are not available, the coordinate
+    variables are used to calculate the projected dimensions.
 
     """
     variable = varinfo.get_variable(variable_name)
-
     range_strings = []
+    if variable.dimensions:
+        range_strings = get_range_strings(variable.dimensions, index_ranges)
+    else:
+        override_dimensions = get_override_projected_dimensions(varinfo, variable_name)
```

Comment:
@sudha-murthy - this is one of the things we discussed this morning, and I wanted to capture a comment on the PR so we don't lose it.

`get_override_projected_dimensions` only returns a list of fake spatial dimensions for the variable. However, for variables that have more dimensions (e.g., 3-D variables), this list is returning a 2-element list for a `numpy` array with 3 axes. This is causing the failure you reported with requests to OPeNDAP. When making requests to OPeNDAP each variable must follow one of two formats in the DAP4 constraint expression:

* Not include _any_ index information for the variable. E.g.: `/variable_name`.
* Including index information for _all_ dimensions. If you want the full range, you can just indicate that with `[]`. So it's something like: `/variable_name[][1:2][3:4]`

Because the returned list of override dimensions only includes spatial information, this function is only constructing part of the list of index information (assuming the example from the second bullet point above is what you want, this function is actually just getting `/variable_name[1:2][3:4]).

The answer here is to use a function that retrieves something for all axes on the variable, and in the correct order. (The naming of the non spatial pseudo-dimensions doesn't matter, because all we're doing is seeing if there is information in the cache of index ranges. If not, the `[]` will be applied for that dimension)

Reply:
This is the logic that is done for a variable with real dimensions:

```
for dimension in variable_dimensions:
    dimension_range = index_ranges.get(dimension)
    if dimension_range is not None and dimension_range[0] <= dimension_range[1]:
        range_strings.append(f'[{dimension_range[0]}:{dimension_range[1]}]')
    else:
        range_strings.append('[]')
```

You can see that if the dimension name is not in the cache of `index_ranges`, then `[]` is used.

Reply:
the override code uses that same method.. But there is no 'dimension name' for the third dimension.

Reply:
Yes - it sounds like we're pretty much agreed on the problem here:

> `get_override_projected_dimensions` only returns a list of fake spatial dimensions for the variable. However, for variables that have more dimensions (e.g., 3-D variables), this list is returning a 2-element list for a `numpy` array with 3 axes.

So the solution is probably to be retrieving the list of variable dimensions in a different way (specifically how is still being hotly debated in a number of places).